dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.test.tsx

```diff
@@ -46,3 +50,46 @@ it("select secrets", () => {
   const selectedCheckbox = totalCheckbox.filterWhere(i => i.prop("checked") === true);
   expect(selectedCheckbox.length).toBe(1);
 });
+
+it("renders the form to create a registry secret", () => {
+  const wrapper = shallow(<AppRepoAddDockerCreds {...defaultProps} />);
+
+  expect(wrapper.text()).not.toContain("Secret Name");
+
+  const link = wrapper.find("a").filterWhere(a => a.text().includes("Add new"));
+  link.simulate("click");
+
+  expect(wrapper.text()).toContain("Secret Name");
+});
+
+it("submits the new secret and re-request the list", async () => {
+  const fetchImagePullSecrets = jest.fn();
+  const createDockerRegistrySecret = jest.fn(() => true);
+  const wrapper = shallow(
+    <AppRepoAddDockerCreds
+      {...defaultProps}
+      fetchImagePullSecrets={fetchImagePullSecrets}
+      createDockerRegistrySecret={createDockerRegistrySecret}
+    />,
+  );
+  const secretName = "repo-1";
+  const user = "foo";
+  const password = "pass";
+  const email = "foo@bar.com";
+  const server = "docker.io";
+  wrapper.setState({ secretName, user, password, email, server, showSecretSubForm: true });
+
+  const button = wrapper.find("button").filterWhere(a => a.text().includes("Submit"));
+  button.simulate("click");
+
+  await wait(1);
```

Comment:
As on the previous PR, keen to know why this hasn't been solved elsewhere without a wait.

Reply:
will reply in the other PR